django/db/models/query.py

```diff
@@ -904,6 +904,9 @@ def _filter_or_exclude(self, negate, *args, **kwargs):
             assert self.query.can_filter(), \
                 "Cannot filter a query once a slice has been taken."
 
+            if self.query.combinator in ['union', 'intersection', 'difference']:
+                raise TypeError('Cannot call .%s() after %s().' %
+                                (['filter', 'exclude'][negate], self.query.combinator))
```

Comment:
Clever use of the integer nature of `bool`, first time I've seen this pattern. It kind of hurts readability though IMO.